vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -318,10 +328,18 @@ def _create_output(
         output.mul_(~after_false_mask).add_(
             recovered_token_ids.mul(after_false_mask))
 
+        # concat spec and nonspec token
+        output_with_bonus_tokens = torch.cat(
+            [output_with_bonus_tokens, non_spec_token_ids])
+
         self.num_accepted_tokens += accepted.sum()
         self.num_emitted_tokens += (output_with_bonus_tokens != -1).sum()
         self.num_draft_tokens += batch_size * k
 
+        # batch_size here maybe 0, as there is case
+        # where no proposal is generated
+        self.num_specs += output_with_bonus_tokens.size()[0]
```

Comment:
Is this right? even if there are no specs this can be nonzero

Reply:
For the ngram case, if we don't match anything, then there would be no proposal.

Reply:
OK. These metrics should work for draft model as well.